### PR TITLE
Added indexes on envelopes

### PR DESCRIPTION
**Pull request by apereverzin**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1227

### Change description ###
Please, refer https://tools.hmcts.net/confluence/display/RBS/SQL+queries+for+envelopes+table for detailed description

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/resources/db/migration/V057__Add_createdat_date_index_on_envelopes.sql [hunk 1]
```diff
@@ -0,0 +1 @@
+CREATE INDEX envelopes_createdat_status_idx ON envelopes((createdat::DATE));
```
Comment (rkondratowicz) [inline src/main/resources/db/migration/V057__Add_createdat_date_index_on_envelopes.sql:1]: should `status` be in the index name?

Comment (apereverzin) [reply src/main/resources/db/migration/V057__Add_createdat_date_index_on_envelopes.sql]: Fixed

File: src/main/resources/db/migration/V057__Add_createdat_date_index_on_envelopes.sql [hunk 2]
```diff
@@ -0,0 +1 @@
+CREATE INDEX envelopes_createdat_idx ON envelopes((createdat::DATE));
```
Comment (doncem) [inline src/main/resources/db/migration/V057__Add_createdat_date_index_on_envelopes.sql:1]: probably don't need extra pair of braces

Comment (apereverzin) [reply src/main/resources/db/migration/V057__Add_createdat_date_index_on_envelopes.sql:1]: there are 2 options: ON envelopes((createdat::DATE)) or ON envelopes(date(createdat))
I prefer the first one

Comment (doncem) [reply src/main/resources/db/migration/V057__Add_createdat_date_index_on_envelopes.sql:1]: ah. ok, makes sense

**Comment by apereverzin** [comment]
Removed index on status, in my local tests it gives some not very significant performance improvement compared to using just index on jurisdiction AND status

**Comment by doncem** [comment]
> Removed index on status, in my local tests it gives some not very significant performance improvement compared to using just index on jurisdiction AND status

perhaps we need to review queries we use and after reviewing them make a decision what's best for db to include certain indices

File: src/main/resources/db/migration/V054__Add_jurisdiction_status_index_on_envelopes.sql [hunk 3]
```diff
@@ -0,0 +1 @@
+CREATE INDEX envelopes_jurisdiction_status_idx ON envelopes(jurisdiction, status);
```
Comment (rkondratowicz) [inline src/main/resources/db/migration/V054__Add_jurisdiction_status_index_on_envelopes.sql:1]: which queries will use this index?

Comment (apereverzin) [reply src/main/resources/db/migration/V054__Add_jurisdiction_status_index_on_envelopes.sql:1]: Please, refer https://tools.hmcts.net/confluence/display/RBS/SQL+queries+for+envelopes+table

**Comment by apereverzin** [comment]
PR-1324 due to conflict in flyway script names

**Pull request closed without merging**